src/main/java/de/mediathekview/mserver/crawler/ndr/tasks/NdrSendungsfolgedetailsTask.java

```diff
@@ -1,120 +1,116 @@
-import org.jsoup.nodes.Document;
-import org.jsoup.nodes.Element;
-import org.jsoup.select.Elements;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
-import com.google.gson.reflect.TypeToken;
 import de.mediathekview.mlib.daten.Film;
+import de.mediathekview.mlib.daten.Resolution;
+import de.mediathekview.mlib.daten.Sender;
+import de.mediathekview.mserver.base.utils.UrlUtils;
+import de.mediathekview.mserver.crawler.ard.json.ArdVideoInfoDTO;
+import de.mediathekview.mserver.crawler.ard.json.ArdVideoInfoJsonDeserializer;
 import de.mediathekview.mserver.crawler.basic.AbstractCrawler;
 import de.mediathekview.mserver.crawler.basic.AbstractDocumentTask;
 import de.mediathekview.mserver.crawler.basic.AbstractUrlTask;
 import de.mediathekview.mserver.crawler.basic.CrawlerUrlDTO;
+import de.mediathekview.mserver.crawler.ndr.NdrConstants;
 import de.mediathekview.mserver.crawler.ndr.parser.NdrFilmDeserializer;
+import de.mediathekview.mserver.crawler.basic.FilmInfoDto;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import mServer.crawler.CrawlerTool;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jsoup.nodes.Document;
 
 public class NdrSendungsfolgedetailsTask extends AbstractDocumentTask<Film, CrawlerUrlDTO> {
+
   private static final Logger LOG = LogManager.getLogger(NdrSendungsfolgedetailsTask.class);
-  private static final String SRC_ARGUMENT = "src";
-  private static final String TIME_PATTERN = "dd.MM.yyyy HH:mm 'Uhr'";
-  private static final String IFRAME_SELECTOR = ".stagePlayer iframe";
-  private static final String TIME_SELECTOR = ".textinfo .subline span:eq(1)";
-  private static final String THEMA_SELECTOR = ".textinfo .subline span:eq(0)";
-  private static final String TITLE_SELECTOR = ".textinfo h1";
   private static final long serialVersionUID = 1614807484305273437L;
-  private static final String ENCODING_GZIP = "gzip";
-  private static final String HEADER_ACCEPT_ENCODING = "Accept-Encoding";
-  private static final Type OPTIONAL_FILM_TYPE_TOKEN = new TypeToken<Optional<Film>>() {}.getType();
-  private final Client client;
+
+  private final NdrFilmDeserializer filmDetailDeserializer;
+  private final Gson gson;
 
   public NdrSendungsfolgedetailsTask(final AbstractCrawler aCrawler,
-      final ConcurrentLinkedQueue<CrawlerUrlDTO> aUrlToCrawlDTOs) {
-    super(aCrawler, aUrlToCrawlDTOs);
-    client = ClientBuilder.newClient();
-    client.register(EncodingFilter.class);
-    client.register(GZipEncoder.class);
-    client.register(DeflateEncoder.class);
-  }
+      final ConcurrentLinkedQueue<CrawlerUrlDTO> aUrlToCrawlDtos) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDtos" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)